src/main/java/com/openshift/internal/restclient/model/PersistentVolume.java

```diff
@@ -0,0 +1,276 @@
+	}
+
+	/**
+	 * @return access modes
+	 * @see {@link com.openshift.restclient.model.volume.PVCAccessModes}
+	 * @see
+	 */
+	@Override
+	public Set<String> getAccessModes() {
+		Set<String> modes = new HashSet<>();
+		ModelNode modelNode = get(PV_ACCESS_MODES);
+		if(!modelNode.isDefined() || !modelNode.getType().equals(ModelType.LIST)) return modes;
+		for (ModelNode node : modelNode.asList()) {
+			modes.add(node.asString());
+		}
+		return modes;
+	}
+
+	/**
+	 * Sets the access modes.
+	 * If there are any modes present, they are overridden by the {@code modes} parameter.
+	 * @param modes
+	 *             the access modes
+	 */
+	@Override
+	public void setAccessModes(String... modes) {
+		ModelNode accessModeNode = get(PV_ACCESS_MODES);
+		if(accessModeNode.isDefined()) {
+			accessModeNode.setEmptyList();
+		}
+		for(String mode : modes) {
+			accessModeNode.add(mode);
+		}
+	}
+
+	/**
+	 * @return the value of 'spec.persistentVolumeReclaimPolicy'
+	 */
+	@Override
+	public String getReclaimPolicy() {
+		return asString(PV_RECLAIM_POLICY);
+	}
+
+	/**
+	 * Sets the 'spec.persistentVolumeReclaimPolicy' value
+	 * @param policy the policy string
+	 */
+	@Override
+	public void setReclaimPolicy(String policy) {
+		get(PV_RECLAIM_POLICY).set(policy);
+	}
+
+	/**
+	 * @return values of 'spec.nfs' in {@link NFSVolume} object
+	 */
+	@Override
+	public NFSVolume getNFSVolume() {
+		ModelNode node = get(PV_NFS);
+		String server = asString(node, SERVER);
+		String path = asString(node, PATH);
+		boolean readOnly = asBoolean(asString(node, READ_ONLY));
+		return new NFSVolume(server, path, readOnly);
+	}
+
+	/**
+	 * Sets the 'spec.nfs' values.
+	 * If any other volume type defined, it's overridden.
+	 * @param volume the {@link NFSVolume} object
+	 */
+	@Override
+	public void setNFSVolume(NFSVolume volume) {
+		removeAllPVTypes();
+		ModelNode node = get(PV_NFS);
+		node.get(SERVER).set(volume.getServer());
+		node.get(PATH).set(volume.getPath());
+		node.get(READ_ONLY).set(volume.isReadOnly());
+	}
+
+	/**
+	 * @return the value of 'spec.hostPath.path'
+	 */
+	@Override
+	public String getHostPathVolume() {
+		ModelNode node = get(PV_HOST_PATH);
+		return asString(node, PATH);
+	}
+
+	/**
+	 * Sets the 'spec.hostPath.path' value of hostPath volume.
+	 * If any other volume type defined, it's overridden.
+	 * @param path the path
+	 */
+	@Override
+	public void setHostPathVolume(String path) {
+		removeAllPVTypes();
+		ModelNode node = get(PV_HOST_PATH);
+		node.get(PATH).set(path);
+	}
+
+	/**
+	 * Deletes all PV types.
+	 * Kubernetes configuration doesn't define consistent definition
+	 * of different PV types. Need to check for all possibilities.
+	 *
+	 * @see 'https://github.com/kubernetes/kubernetes/issues/16155'
+	 * @see 'https://github.com/kubernetes/kubernetes/blob/a10ed3e2b178096dd9575874f7e5916ff244dcea/docs/user-guide/volumes.md#types-of-volumes'
+	 *
+	 */
+	private void removeAllPVTypes() {
```

Comment:
I dont think this will be necessary if impl. a single volume source as defined in the kube types: https://github.com/kubernetes/kubernetes/blob/master/pkg/api/types.go#L169  Most of the patterns upstream are to have a 'type' field to tell you what type the actual volume is.  The expectation is that only one kind will be populated while the others are empty.